### PR TITLE
Us 256992 damage api

### PR DESCRIPTION
**Pull request by SzymonSmykala**

### CODE REVIEW

File: iOS/SampleApp/SampleApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -89,6 +91,7 @@
 		B52DB80921BA8EEF00C6CC36 = {
 			isa = PBXGroup;
 			children = (
+				D74F2B612221AF110055ACC1 /* DemagePartsServiceFactory.swift */,
```
[the remark this answers is not in the document]
Comment (SzymonSmykala) [reply iOS/SampleApp/SampleApp.xcodeproj/project.pbxproj]: to be deleted

File: iOS/SampleApp/SampleApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -222,6 +225,7 @@
 			files = (
 				B52DB81821BA8EEF00C6CC36 /* ViewController.swift in Sources */,
 				B52DB81621BA8EEF00C6CC36 /* AppDelegate.swift in Sources */,
+				D74F2B622221AF120055ACC1 /* DemagePartsServiceFactory.swift in Sources */,
```
[the remark this answers is not in the document]
Comment (SzymonSmykala) [reply iOS/SampleApp/SampleApp.xcodeproj/project.pbxproj]: to be deleted

File: iOS/FVM/FVM/FVMCarModelViewController.swift [hunk 3]
```diff
@@ -19,15 +19,59 @@ public class FVMCarModelViewController : SCNView {
     internal var scnScene: SCNScene!
     internal var scnCamera: SCNNode!
     internal let highlightHandler = HighlightHandler()
+    private var damagePartsService: DamagedPartsService?
     
+    private let simpleJson = """
```
[the remark this answers is not in the document]
Comment (SzymonSmykala) [reply iOS/FVM/FVM/FVMCarModelViewController.swift]: proof of concept, it isn't whole US

File: iOS/FVM/FVM/DamagedParts/Factories/DamagePartsServiceFactory.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+public class DamagePartsServiceFactory{
+    
+    public static func Create(validPartsNames: [String]) -> DamagedPartsService{
```
Comment (ghawliczek) [inline iOS/FVM/FVM/DamagedParts/Factories/DamagePartsServiceFactory.swift]: is this a convention? Static method names written with capital letter? If not, change it to create(...)

File: iOS/FVM/FVM/DamagedParts/Services/DamagePartService.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+public protocol DemagedPartsServiceProtocol {
+    func createAndGetCollectionOfDamagedParts(json: String) -> [Selection]
```
Comment (ghawliczek) [inline iOS/FVM/FVM/DamagedParts/Services/DamagePartService.swift]: Is this method necessary in the interface definition?

**Comment by SzymonSmykala** [comment]
> I have noticed you sometimes set access modifier in classes and function which are for testing or sometimes not. Could you decide if it is needed or not and fix it ?
> 
> Also I think access to your damage API classes is too big. If you agree with me, I would like you to appropriate diminish it.

**Pull request closed without merging**